transport/ws/src/main/java/org/kaazing/gateway/transport/ws/extension/WebSocketExtensionFactory.java

```diff
@@ -62,7 +89,7 @@ private WebSocketExtensionFactory(Map<String, WebSocketExtensionFactorySpi> fact
      * @throws ProtocolException  If the extension cannot be negotiated. Throwing this exception will result
      *                      in failing the WebSocket connection.
      */
-    public WebSocketExtensionSpi negotiate(ExtensionHeader extension, WsResourceAddress address) throws ProtocolException {
+    public WebSocketExtension negotiate(ExtensionHeader extension, WsResourceAddress address) throws ProtocolException {
```

Comment:
This method does not need to be public I think. You can remove the "public" qualifier, leaving it package private so it can be called by the unit tests